decidim-proposals/app/models/decidim/proposals/proposal.rb

```diff
@@ -205,6 +205,11 @@ def editable_by?(user)
         !answered? && within_edit_time_limit? && !copied_from_other_component? && created_by?(user)
       end
 
+      def promotable_by?(user)
+        not_likely_to_be_promoted = Proposal.where(title: title).count == 1
```

Comment:
Thanks. The intention is to disallow promoting twice. As promoting a rejected emendation creates a new amendable type (proposal) with one click, and without using the create form, it doesnt do all the checks that those forms do (for example, proposals have lots of steps in the create process). So if we let the user to click more than once, new identical propoposals will be created.

This solution isnt the prettiest but is the one I found at the moment. When promoting, there is already one amendable type (proposal) with the same title, which is the rejected emendation. After promoting, there are two amendable types with the same title so it makes `not_likely_to_be_promoted` false and hides the button.

There are occasions were this cannot work, for example, you can create an amendment with the same title as OTHER proposal (not the one you're amending, that is forbidden). If this amendment is rejected the promote button won't show because `not_likely_to_be_promoted` will be false from the start. This could be fixed by redirecting promote to the create form of the amendable type, but I was told it was with one click.

Reply:
The brittle method promotable_by? has been refactored thanks to your suggestions. Let me know if it can be improved: https://github.com/decidim/decidim/pull/3986/commits/752f66ae646805ad6742ac5267718c65a53abc30 (https://github.com/decidim/decidim/pull/3986/commits/8bc92094120f4471c319aec12020f075d9506a85 forgot to check with rubocop)

Reply:
Thanks. I've refactored the permissions to promote (now in the amendments_helper). Let me know if it can be improved:
https://github.com/decidim/decidim/blob/8c4df8a4ecedf9e90f1b9f49d777dcd9424b43d8/decidim-core/app/helpers/decidim/amendments_helper.rb#L98